### PR TITLE
JIT: fix prejit entry PGO instrumentation

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Don't assume the entry block is `fgFirstBB`. Make sure to put the
entry probe in the scratch bb.

Addresses #49108.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@clamp03 see if this fixes the issues you were seeing.

cc @dotnet/jit-contrib

**Comment by clamp03** [comment]
@AndyAyersMS Thank you. I will test and let you know.

**Comment by clamp03** [comment]
@AndyAyersMS It works well. Thank you so much!

File: src/coreclr/jit/fgprofile.cpp [hunk 1]
```diff
@@ -407,11 +419,11 @@ void BlockCountInstrumentor::InstrumentMethodEntry(Schema& schema, BYTE* profile
 
     // Find the address of the entry block's counter.
     //
-    BasicBlock* const block            = m_comp->fgFirstBB;
-    const int         firstSchemaIndex = block->bbCountSchemaIndex;
-    assert(block->bbCodeOffs == (IL_OFFSET)schema[firstSchemaIndex].ILOffset);
+    const int firstSchemaIndex = (int)m_entryBlock->bbCountSchemaIndex;
+    assert(m_entryBlock->bbCodeOffs == (IL_OFFSET)schema[firstSchemaIndex].ILOffset);
```
Comment (briansull) [inline src/coreclr/jit/fgprofile.cpp:423]: Wouldn't this be clearer if we asserted that both are zero?

```
assert(m_entryBlock->bbCodeOffs == 0);
assert(schema[firstSchemaIndex].ILOffset == 0);
```

Comment (briansull) [reply src/coreclr/jit/fgprofile.cpp:423]: Also do we need to assert that `m_entryBlock` is not nullptr?

**Review by briansull — APPROVED** [review]
Looks Good with two comments

File: src/coreclr/jit/fgprofile.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/fgprofile.cpp]: Thanks -- made updates.

**Comment by AndyAyersMS** [comment]
MUSL failure is some sort of corrputed download, will ignore.
```
Extraction failed for file: /datadisks/disk1/workspace/_work/1/s/__download__/libraries_test_assets_Linux_x64_Debug/libraries_test_assets_Linux_x64_Debug.tar.gz 
```

**Pull request merged**